search/search-client-elasticsearch/src/main/java/io/camunda/search/es/transformers/aggregator/SearchAggregationResultTransformer.java

```diff
@@ -100,4 +180,24 @@ private Map<String, AggregationResult> transformAggregation(
         });
     return result;
   }
+
+  private List<SearchQueryHit> toSearchQueryHits(
+      final List<Hit<JsonData>> hits, final Class<T> documentClass) {
+    if (hits != null) {
+      final var hitTransformer = new SearchQueryHitTransformer<T>(transformers);
+      return hits.stream()
+          .filter(hit -> Objects.nonNull(hit.source()))
+          .map(
+              hit ->
+                  new Hit.Builder<T>()
+                      .index(hit.index())
+                      .id(hit.id())
+                      .score(hit.score())
+                      .source(hit.source().to(documentClass))
+                      .build())
+          .map(hitTransformer::apply)
+          .collect(Collectors.toList());
+    }
+    return new ArrayList<>();
```

Comment:
🙃 Can we use something like `List.of()` here? Or does it have to be a mutable list?